### ISSUE TITLE
Min_delta parameters on callbacks function of Keras model seems to not take effect

### ISSUE THREAD

**Issue by NikosSpanos** [tensorflow/tensorflow#41417]
I am using the following callbacks function on a Keras model and I initialize the minimum delta to 0.002, so based on the documentation of Tensorflow/Keras any improvement in the validation loss function less than 0.002 won't be counted for an improvement. However, this seems to not get implemented in my case.

callback function:

```
def callback(folder_path, saved_model_name, patience_value, logdir, hparams):
    
    # Initialize parameters
    monitor_metric = 'val_loss'
    minimum_delta = 0.002
    patience_limit = patience_value
    verbose_value = 1
    mode_value = 'min'
    weights_fname = os.path.join(os.getcwd(), '{0}/{1}.h5'.format(folder_path, saved_model_name))
    
    # Initialize callbacks
    callbacks = [
        
        EarlyStopping(monitor=monitor_metric,
                      min_delta=minimum_delta,
                      patience=patience_limit,
                      verbose=verbose_value,
                      mode=mode_value,
                      restore_best_weights=True),

        ModelCheckpoint(filepath=weights_fname,
                        monitor=monitor_metric,
                        verbose=verbose_value,
                        save_best_only=True,
                        save_weights_only=True),
        
        tf.keras.callbacks.TensorBoard(logdir),
        
        hp.KerasCallback(logdir, hparams)
    ]
    
    return callbacks
```
**Example 1**
```
Epoch 00016: val_loss improved from 0.02129 to 0.02015, saving model to /content/model_one/adam_v2_14072020/multi_input_keras_model_50dim_128batchsize_0.01lr_10decaymultiplier_14072020.h5
```

**Example 2**
```
Epoch 00019: val_loss improved from 0.01880 to 0.01803, saving model to /content/model_one/adam_v2_14072020/multi_input_keras_model_50dim_128batchsize_0.01lr_10decaymultiplier_14072020.h5
```
You can see that in both examples the validation loss improved and the model's weight have been saved, while they shouldn't;t have:

* 0.02129 - 0.02015 = 0.00114 < 0.002 (although it was counted as an improvement)
* 0.01880 - 0.01803 = 0.00077 < 0.002 (also counted as an improvement in validation loss)
What is going wrong?

Apologize for not posting more code since it's a thorough application. So, please check my [colab notebook](https://drive.google.com/file/d/1FqQjjB2xFFQCiAqZacvVsl5SU0kcJwGd/view?usp=sharing) to replicate the issue.

Pickled data saved on my drive [folder](https://drive.google.com/drive/folders/1Z0p2RVpJSn8uhZG5UvZBU1AS3aLqEVqv?usp=sharing)

**Comment by ravikyram** [comment]
@NikosSpanos 

Can you please provide supporting files as well (Eg: saved pickle files) to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by NikosSpanos** [comment]
> @NikosSpanos
> 
> Can you please provide supporting files as well (Eg: saved pickle files) to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

Ofc @ravikyram , I added the relative folder at the end of my post

**Comment by gowthamkpr** [comment]
@NikosSpanos Is this happening with just this model or is it happening with other models too. Can you please try this in a basic example and also using tf-night and let me know i the issue still persists. Thanks!

**Comment by NikosSpanos** [comment]
@gowthamkpr Can you please validate that my statement is valid? Because I have posted in the past the same question on StackOverflow (link [here](https://stackoverflow.com/questions/62912769/min-delta-on-callbacks-function-of-keras-model-seems-to-not-take-effect). 
Based on the answers given in the posted SO question, I figured out that the minimum_delta is not the difference between validation_loss of epochs, but rather is the number below which if no improvement is spotted for 10 consecutive epochs then the training stops. So in my posted example above, since minimum delta=0.002 and validation loss: 0.01803 > 0.002. The training continuous until validation loss < 0.002

Can you validate this statement. If you verify that my intuition is wrong, I will post a simpler example of the problem. Also, how am I suppose to use tf-nightly? I use TensorFlow v2.2

**Comment by gowthamkpr** [comment]
@NikosSpanos As answered in stack overflow and as mentioned in the [documentation](https://keras.io/api/callbacks/early_stopping/), its pretty clear that if you don't see the change (or improvement of min_delta) for the number of epochs of patience_value, the training stops.

This has nothing to do with ModelCheckpoint.
